src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -1,143 +1,199 @@
 import { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { PropTypes } from 'prop-types';
 import { v4 as uuidv4 } from 'uuid';
+
 import Input from '../../common/Input/Input';
 import Button from '../../common/Button/Button';
 import Textarea from '../../common/Textarea/Textarea';
 import { mockedAuthorsList } from '../../constants';
 import pipeDuration from '../../helpers/pipeDuration';
 import './createCourse.css';
 
-function CreateCourse({ courseData, setCourseData, handleSubmit }) {
+function CreateCourse({ courses, setCourses }) {
 	const [authorName, setAuthorName] = useState('');
-
 	const [authors, setAuthors] = useState(mockedAuthorsList);
-	const authorsList = authors.map((author) => {
-		return (
-			<div className='author' key={author.id}>
-				<p>{author.name}</p>
-				<Button
-					buttonText='Add author'
-					type='button'
-					handleClick={() => {
-						setCourseData({
-							...courseData,
-							authors: [
-								...courseData.authors,
-								{ id: author.id, name: author.name },
-							],
-						});
-						setAuthors((prev) => [
-							...prev.filter((item) => !(item.id === author.id)),
-						]);
-					}}
-				/>
-			</div>
-		);
-	});
+	const [courseTitle, setCourseTitle] = useState('');
+	const [courseDescription, setCourseDescription] = useState('');
+	const [courseAuthors, setCourseAuthors] = useState([]);
+	const [courseDuration, setCourseDuration] = useState('');
+	const navigate = useNavigate();
+
+	const handleAddAuthor = (author) => {
+		setCourseAuthors([...courseAuthors, { id: author.id, name: author.name }]);
+		setAuthors([...authors.filter((item) => !(item.id === author.id))]);
```

Comment:
`filter` returns a new array, so array destructing is unnecessary here
```suggestion
		setAuthors(authors.filter((item) => !(item.id === author.id)));
```

Reply:
also, you can use `!==` operator here